tests/tests/tx.rs

```diff
@@ -475,6 +464,99 @@ async fn get_transactions_by_owner_forward_and_backward_iterations() {
     );
 }
 
+#[tokio::test]
+async fn get_transactions_by_owner_backward_iterations() {
+    let alice = Address::from([1; 32]);
+    let bob = Address::from([2; 32]);
+
+    let mut context = TestContext::new(100).await;
+    let _ = context.transfer(alice, bob, 1).await.unwrap();
+    let _ = context.transfer(alice, bob, 2).await.unwrap();
+    let _ = context.transfer(alice, bob, 3).await.unwrap();
+    let _ = context.transfer(alice, bob, 4).await.unwrap();
+    let _ = context.transfer(alice, bob, 5).await.unwrap();
+
+    let client = context.client;
+
+    let all_transactions_backward = PaginationRequest {
+        cursor: None,
+        results: 10,
+        direction: PageDirection::Backward,
+    };
+    let response = client
+        .transactions_by_owner(&bob, all_transactions_backward)
+        .await
+        .unwrap();
+    let transactions_backward = response
+        .results
+        .into_iter()
+        .map(|tx| {
+            assert!(matches!(tx.status, TransactionStatus::Success { .. }));
+            tx.transaction
+        })
+        .collect_vec();
+    assert_eq!(transactions_backward.len(), 5);
```

Comment:
This looks like you're testing that the request returns paginated data with the correct length. After this, you test that the request returns paginated data with the expected results. Lastly, you test that the request returns paginated data with the expected results when using a cursor. Therefore, I think this test could be broken up into two or three separate tests, since these are three separate concerns.

Reply:
Splitting this text into three doesn't make much sense because the third test will be almost the same except for two asserts. It increases the test file size and makes it harder to navigate by adding a very slight improvement to the one test readability.

I agree that we need to try to test different things with different tests, but when one test is the copy of another + one new line, it is not very nice.

Reply:
This is my opinion on testing, with a bias from working in TDD:

Each unit test should illustrate the conformance of a single, functional behaviour (a unit). The behaviour being tested should be described succinctly by the test name. The test suite creates a specification that outlines the complete behaviour of the subject under test, and this specification acts as a promise to the developers using the code: The test names tell us what it will do and the test body proves that it does it. A well-written test suite can serve as living documentation for the code. It clearly demonstrates the intended use cases and the expected behaviour of the software. E.g., `"backward_pagination_returns_correct_number_of_results"`, "`backward_pagination_returns_results_in_reverse_order`", `"backward_pagination_supports_cursor`", etc. 

Therefore, the goal isn't simply readability, and the fact that test code is duplicated is not really important. What's more important is that the test suite communicates the intended behaviour succinctly, and test names are the easiest way to achieve that. When I read the test mod, or run all the tests, I can simply look at the list of test names to have an overall sense of what the behaviours are.

If that resonates with you, feel free to update the test suite. If not, no worries - again, this is just my opinion.

Reply:
https://github.com/FuelLabs/fuel-core/pull/1492/commits/ef6cfc6964bd486b267b72f9ac9b52041753149a